include/sys/spa.h

```diff
@@ -408,6 +488,26 @@ _NOTE(CONSTCOND) } while (0)
 #define	BP_GET_LEVEL(bp)		BF64_GET((bp)->blk_prop, 56, 5)
 #define	BP_SET_LEVEL(bp, x)		BF64_SET((bp)->blk_prop, 56, 5, x)
 
+/* encrypted, authenticated, and MAC cksum bps use the same bit */
+#define	BP_USES_CRYPT(bp)		BF64_GET((bp)->blk_prop, 61, 1)
+#define	BP_SET_CRYPT(bp, x)		BF64_SET((bp)->blk_prop, 61, 1, x)
+
+#define	BP_IS_ENCRYPTED(bp)			\
+	(BP_USES_CRYPT(bp) &&			\
+	BP_GET_LEVEL(bp) <= 0 &&		\
+	DMU_OT_IS_ENCRYPTED(BP_GET_TYPE(bp)))
```

Comment:
Per https://github.com/openzfs/openzfs/pull/124#issuecomment-305624736 , it would be better if the BP layout didn't depend on the `dmu_ot` table.  For example, you need `dmu_ot` to tell whether dva[2] is a DVA vs salt+IV.  Could we say that AUTHENTICATED BP's share the same layout as ENCRYPTED BP's, but their salt and IV are zero?